include/qemu/log_instr.h

```diff
@@ -104,13 +104,14 @@ typedef struct {
     bool force_drop;
     /* We are starting to log at the next commit */
     bool starting;
-    /*
-     * Opaque handle to the current instruction info
-     * TODO(am2419): It would be interesting to have a ring buffer
-     * of log_instr_info here, so that we can avoid dumping to file
-     * all the time.
-     */
-    struct cpu_log_instr_info *instr_info;
+    /* Per-CPU flags */
+    int flags;
+#define QEMU_LOG_INSTR_FLAG_BUFFERED 1
+
+    /* Ring buffer of log_instr_info */
+    GArray *instr_info;
```

Comment:
Do we need the extra indirection via GArray? Since QEMU_LOG_INSTR_CPU_BUF_ENTRIES is a constant, we could just use `struct cpu_log_instr_info *instr_info[QEMU_LOG_INSTR_CPU_BUF_ENTRIES]`?

Yes, it does waste about 64K*sizeof(void*) of memory when tracing without a ring buffer, but should make the ring buffer case faster?

Reply:
I initially did this because I was not thinking about statically sizing it. For now we can probably get away with the static size until we have the need to change it.